utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/source/UniversalKafkaSourceFactory.scala

```diff
@@ -121,8 +137,42 @@ class UniversalKafkaSourceFactory[K: ClassTag: NotNothing, V: ClassTag: NotNothi
     // prepare KafkaDeserializationSchema based on given key and value schema (without schema evolution - we want format test-data exactly the same way, it was sent to kafka)
     val formatterSchema = schemaBasedMessagesSerdeProvider.deserializationSchemaFactory.create[K, V](kafkaConfig, None, None)
     val recordFormatter = schemaBasedMessagesSerdeProvider.recordFormatterFactory.create[K, V](kafkaConfig, formatterSchema)
+    implProvider.createSource(params, dependencies, finalState.get, List(preparedTopic), kafkaConfig, deserializationSchema,
+      recordFormatter, kafkaContextInitializer, getParameters(valueSchemaUsedInRuntime, preparedTopic.original))
+  }
 
-    implProvider.createSource(params, dependencies, finalState.get, List(preparedTopic), kafkaConfig, deserializationSchema, recordFormatter, kafkaContextInitializer)
+  private def getParameters(runtimeSchema: Option[RuntimeSchemaData[ParsedSchema]], topic: String)
+                           (implicit nodeId: NodeId): KafkaTestParametersInfo = {
+    Validated.fromOption(runtimeSchema, NonEmptyList.one(CustomNodeError(nodeId.id, "Cannot generate test parameters: no runtime schema found", None)))
+      .andThen(handleSchemaType(_, topic))
+      .valueOr(e => throw new RuntimeException(e.toList.mkString("")))
+  }
+
+  private def handleSchemaType(determinedSchema: RuntimeSchemaData[ParsedSchema], topic: String)(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, KafkaTestParametersInfo] = {
+    (determinedSchema.schema match {
+      case s: AvroSchema => AvroSinkValueParameter(s.rawSchema()).map(_.toParameters)
```

Comment:
We have a concept that all switches on schema are in `UniversalSchemaSupportDispatcher` so it will be easy to add support for another type of schema. Can you move this there?

Reply:
Not so easy :). Take a look how it is done there. You should have each schema handling in each *SchemaSupport (`AvroSchemaSupport`, `JsonSchemaSupport`)